src/unabto/unabto_tcp_fallback.c

```diff
@@ -125,6 +125,12 @@ void unabto_tcp_fallback_socket_closed(nabto_connect* con) {
     if (con->state == CS_CONNECTED && con->type == NCT_REMOTE_RELAY_MICRO) {
         con->state = CS_CONNECTING;
         con->conAttr = CON_ATTR_DEFAULT;
+        if (con->relayIsActive) {
+            NABTO_LOG_DEBUG((PRI_tcp_fb "closing full connection as relay was active when closed", TCP_FB_ARGS(con)));
+            nabto_release_connection_req(con);
```

Comment:
This seems ok as unabto_tcp_fallback.c seems to handle the tcp fallback specific connection logic even that the state is defined in the nabto_connect struct.